api/apps/api/src/modules/access-control/scenarios-acl/scenario-acl.service.ts

```diff
@@ -104,4 +108,54 @@ export class ScenarioAclService implements ScenarioAccessControl {
     });
     return otherOwnersInScenario >= 1;
   }
+
+  async updateUserInScenario(
+    scenarioId: string,
+    userAndRoleToChange: UserRoleInScenarioDto,
+    loggedUserId: string,
+  ): Promise<Either<Permit, void>> {
+    const { userId, roleName } = userAndRoleToChange;
+    if (!(await this.isOwner(loggedUserId, scenarioId))) {
+      return left(false);
+    }
+    assertDefined(roleName);
+    const apiDbConnection = getConnection(DbConnections.default);
+    const apiQueryRunner = apiDbConnection.createQueryRunner();
+
+    await apiQueryRunner.connect();
+    await apiQueryRunner.startTransaction();
+
+    try {
+      const existingUserInScenario = await apiQueryRunner.manager.findOne(
+        UsersScenariosApiEntity,
+        undefined,
+        {
+          where: {
+            scenarioId,
+            userId,
+          },
+        },
+      );
+
+      if (!existingUserInScenario) {
+        const userRoleToSave = new UsersScenariosApiEntity();
+        userRoleToSave.scenarioId = scenarioId;
+        userRoleToSave.userId = userId;
+        userRoleToSave.roleName = roleName;
+
+        await apiQueryRunner.manager.save(userRoleToSave);
+      } else {
+        await apiQueryRunner.manager.update(
+          UsersScenariosApiEntity,
+          { scenarioId, userId },
+          { roleName },
+        );
+      }
+    } catch (err) {
+      await apiQueryRunner.rollbackTransaction();
```

Comment:
so if one of the queries above fails, an exception is thrown, but you still return right(), which seems off to me. Shouldn't we inform the user that something failed?

Reply:
After fighting with debugging in VSCode for the try/catch/finally I think now it handles better this part, both by informing the user if an error occurs and returning `right()` in the right place.